### PR TITLE
Add mock of persistent cache in:

### PR DESCRIPTION
**Pull request by muratovv**
Add useful mock for tx persistent cache.

Mock is added to test and irohad/application folders for easy access.

Signed-off-by: Fedor Muratov <muratovfyodor@yandex.ru>

### CODE REVIEW

File: irohad/main/application.cpp [hunk 1]
```diff
@@ -41,6 +42,42 @@ using namespace iroha::consensus::yac;
 
 using namespace std::chrono_literals;
 
+// TODO: @muratovv 12.11.2018 remove the mock after effective implementation
+// will be finished IR-1840
+#include <gmock/gmock.h>
```
Comment (kamilsa) [inline irohad/main/application.cpp]: weird place to put include

File: test/module/irohad/ametsuchi/ametsuchi_mocks.hpp [hunk 2]
```diff
@@ -336,7 +338,39 @@ namespace iroha {
                    bool(const shared_model::interface::BlocksQuery &));
     };
 
+    class MockTxPresenceCahce : public iroha::ametsuchi::TxPresenceCache {
```
Comment (nickaleks) [inline test/module/irohad/ametsuchi/ametsuchi_mocks.hpp]: typo in the class name

File: test/module/irohad/ametsuchi/ametsuchi_mocks.hpp [hunk 3]
```diff
@@ -336,7 +338,39 @@ namespace iroha {
...
+      MOCK_CONST_METHOD1(
+          check,
+          iroha::ametsuchi::TxPresenceCache::BatchStatusCollectionType(
+              const shared_model::interface::TransactionBatch &));
+    };
+
   }  // namespace ametsuchi
 }  // namespace iroha
 
+namespace std {
```
Comment (nickaleks) [inline test/module/irohad/ametsuchi/ametsuchi_mocks.hpp]: overloading things in std namespace is undefined behavior. If you want google test to print your values, please refer to: https://github.com/google/googletest/blob/master/googletest/docs/advanced.md#teaching-googletest-how-to-print-your-values

It has 2 solutions: define << operator, or PrintTo function. I would prefer PrintTo as it is more google test specific.

However if you decide to overload <<, then do it in the same namespace as the classes you have declared.

File: irohad/main/application.cpp [hunk 4]
```diff
@@ -41,6 +42,42 @@ using namespace iroha::consensus::yac;
...
+class MockTxPresenceCahce : public iroha::ametsuchi::TxPresenceCache {
```
Comment (kamilsa) [inline irohad/main/application.cpp]: Fix typo in the class's name

File: test/module/irohad/ametsuchi/ametsuchi_mocks.hpp (hunk 2 above)
Comment (kamilsa) [inline test/module/irohad/ametsuchi/ametsuchi_mocks.hpp]: Again, fix typo

File: irohad/main/application.cpp (hunk 1 above)
Comment (muratovv) [reply irohad/main/application.cpp]: This was left intentionally

**Pull request merged**